### PR TITLE
Feature tictactoe

### PR DESCRIPTION
**Pull request by pipe2442**
Tic-tac-toe - Javascript

This branch was created to merge tic tac toe feature and functions with javascript.

- Created Html, CSS and JavaScript base.
- Created functions to store board in an array and loop the game.
- Created buttonto start the game and restart when win casses occur.
- Add styling with css and linters with stylelintrc.json.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-mixed-operators, no-use-before-define */
+let game = false;
+let turnNumber = 0;
+let board;
+const oPlayer = 'o';
+const xPlayer = 'x';
+const cells = document.querySelectorAll('.cell');
```
Comment (SajjadAhmad14) [inline app.js]: Please try avoiding global code.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    if (turnNumber % 2 === 0) {
+      if (board[e.target.id] === oPlayer || board[e.target.id] === xPlayer) {
+        document.getElementById('win').innerHTML = '<h1>That\'s not a valid move!!!</h1>';
+        turnNumber -= 1;
+      } else {
+        turn(e.target.id, oPlayer);
+      }
+    } else if (turnNumber % 2 !== 0) {
+      if (board[e.target.id] === oPlayer || board[e.target.id] === xPlayer) {
+        document.getElementById('win').innerHTML = '<h1>That\'s not a valid move!!!</h1>';
```
Comment (SajjadAhmad14) [inline app.js]: DOM manipulation should not be mixed up with app logic.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [ ] Game UI is very appealing.
- [ ] You have mastered Git flow.
- [ ] PR title and summary are descriptive.
- [ ] README provides all the necessary info about the repository.

## To Improve 🙏
- [ ] Please see inline comments for reference.
- [ ] You should not use global code. Please encapsulate logic inside modules or factories. See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/global_requirement.png).
- [ ] Please use game, gameBoard, player objects. See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/objects_requirement.png).
- [ ] Game should allow players to enter their names.
- [ ] Please uses an array inside of a gameBoard object to store the 'moves'.
- [ ] Players should be stored as objects. See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/objects_requirement.png).
- [ ] Application logic should be separated from DOM manipulation tasks.
# Optional
- [ ] You can use [Netlify](https://www.netlify.com/) to host your app.
### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Comment by pipe2442** [comment]
Thanks for the previews review, changes has been implemented.

**Review by MkrtichSargsyan — CHANGES_REQUESTED** [review]
## _Status:_ Changes Requested :recycle:

## Hello Guys

THIS IS VERY IMPRESSIVE WORK. YOU HAVE DONE A GOOD JOB :+1:

## To Highlight :tada:


1. Your PR has a good title and description.
2. You have descriptive git commit messages.


Good job on implementing the changes requested. However, there are some changes that you need to make in order to get your project approved

- [ ] It has game, gameBoard, player objects [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/global_requirement.png)

- [x] It doesn't show if the move is a tie

err


- [x] Please validate user input, so the usernames should not be empty

- [x] Don't disable that much eslint-rules


<hr>

## OPTIONAL

- [ ] Kindly find a way to make the checkWinner function cleaner

Please, make the REQUIRED changes and submit your request again! :grinning:
Good luck. 

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

File: app.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    && board[1].innerText === board[4].innerText && board[1].innerText === board[7].innerText
+    || (a === board[2].innerText || b === board[2].innerText)
+    && board[2].innerText === board[5].innerText && board[2].innerText === board[8].innerText
+    || (a === board[0].innerText || b === board[0].innerText)
+    && board[0].innerText === board[4].innerText && board[0].innerText === board[8].innerText
+    || (a === board[2].innerText || b === board[2].innerText)
+    && board[2].innerText === board[4].innerText && board[2].innerText === board[6].innerText) {
+      if (turnNumber % 2 === 0) {
+        document.getElementById('win').innerHTML = `<h1>${player2.name.value} wins</h1>`;
+      } else if (turnNumber % 2 !== 0) {
```
Comment (MkrtichSargsyan) [inline app.js]: No need for else if, just write else

File: app.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      document.getElementById('win').innerHTML = `<h1>It's ${next.name.value} turn</h1>`;
+      cell.innerText = player.marker;
+    }
+  };
+
+  let turnNumber = 0;
+  marker = (e) => {
+    if (turnNumber % 2 === 0) {
+      turn(cells[e.target.id], player2, player1);
+    } else if (turnNumber % 2 !== 0) {
```
Comment (MkrtichSargsyan) [inline app.js]: instead of else if you can write else without any condition

**Comment by pipe2442** [comment]
Thanks for the review, the project was fixed with requested changes.

**Review by bellom — APPROVED** [review]
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**